hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SstFilteringService.java

```diff
@@ -147,10 +147,9 @@ private void markSSTFilteredFlagForSnapshot(String volume, String bucket,
     @Override
     public BackgroundTaskResult call() throws Exception {
 
-      Optional<SnapshotCache> snapshotCache = Optional.ofNullable(ozoneManager)
-          .map(OzoneManager::getOmSnapshotManager)
-          .map(OmSnapshotManager::getSnapshotCache);
-      if (!snapshotCache.isPresent()) {
+      Optional<OmSnapshotManager> snapshotManager = Optional.ofNullable(ozoneManager)
+          .map(OzoneManager::getOmSnapshotManager);
+      if (!snapshotManager.isPresent()) {
```

Comment:
It will be evaluated to false even if snapshot feature is disabled because snapshotManager obj is not null in that case. We should check if snapshot feature is enabled and has some snapshots.

Reply:
Actually you are right, it will be only null when snapshot feature is disabled. The original check is kind of confusing and hides when it is actually there.
https://github.com/apache/ozone/pull/5008/files#r1258733466

You can leave it as it is in the patch or change it to actual use feature flag. I'm fine either way.